.github/workflows/verify.yaml

```diff
@@ -0,0 +1,89 @@
+name: verify
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.21'
+
+      - name: Install ShellCheck and shfmt
+        run: |
+          sudo apt update
+          sudo apt install -y shellcheck
+          go install mvdan.cc/sh/v3/cmd/shfmt@latest
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          args: --timeout=5m --out-format=colored-line-number
+          only-new-issues: true
+
+      - name: codespell
+        uses: codespell-project/actions-codespell@v2
+        with:
+          skip: vendor,*.svg
+
+      - name: ShellCheck for wait_for_deployment.sh
+        run: shellcheck ./test/wait_for_deployment.sh
+
+      - name: shfmt for wait_for_deployment.sh
+        run: shfmt -d ./test/wait_for_deployment.sh
+
+  gomod:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.21'
+
+      - name: verify go.mod/go.sum
+        run: |
+          make vendor
+          git diff --exit-code
+
+      - name: Upload go.sum diff if exists
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: go-sum-diff
+          path: go.sum.diff
```

Comment:
no need to have this; CI job failure logs are enough to find out what's wrong.

Reply:
@kolyshkin thank you for your review. To be honest I haven't worked with GitHub workflows that much previously so I think did some mistakes here unintentionally.